helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -126,4 +123,53 @@ public OptimalAssignment calculate(ClusterModel clusterModel) throws HelixRebala
     return hardConstraints.stream().map(HardConstraint::getDescription)
         .collect(Collectors.toList());
   }
+
+  // TODO investigate better ways to sort replicas. One option is sorting based on the creation time.
+  private List<AssignableReplica> getOrderedAssignableReplica(ClusterModel clusterModel) {
+    Map<String, Set<AssignableReplica>> replicasByResource = clusterModel.getAssignableReplicaMap();
```

Comment:
nit: Add the get method in cluster model to get all assignableReplicas. I feel the same code is repeated elsewhere

Reply:
I think this is the only one. The others are all tests.